build-tools-internal/performance/elasticsearch-build-tool-update.scenarios

```diff
@@ -0,0 +1,84 @@
+# Can specify scenarios to use when none are specified on the command line
+default-scenarios = ["buildConfiguration_master", "buildConfiguration_branch", "single_project_branch", "single_project_master", "precommit_master", "precommit_branch"]
+
+buildConfiguration_branch {
+    title = "configuration phase (@testGitCommit@)"
+    tasks = ["help"]
+    gradle-args = ["--no-scan", "--no-build-cache"]
+    run-using = cli // value can be "cli" or "tooling-api"
+    daemon = warm // value can be "warm", "cold", or "none"
+    warm-ups = 5
+    iterations = 10
+    git-checkout = {
+        build = "@testGitCommit@"
+    }
+}
+
+buildConfiguration_master {
+    title = "configuration phase (master)"
+    tasks = ["help"]
+    gradle-args = ["--no-scan", "--no-build-cache"]
+    run-using = cli // value can be "cli" or "tooling-api"
+    daemon = warm // value can be "warm", "cold", or "none"
+    warm-ups = 5
+    iterations = 10
+    git-checkout = {
+        build = "master"
+    }
+}
+
+precommit_branch {
+    title = "precommit (@testGitCommit@)"
+    cleanup-tasks = ["clean"]
+    tasks = ["precommit"]
+    gradle-args = ["--no-scan", "--no-build-cache"]
+    run-using = cli // value can be "cli" or "tooling-api"
+    daemon = warm // value can be "warm", "cold", or "none"
+    warm-ups = 5
+    iterations = 10
+    git-checkout = {
+        build = "@testGitCommit@"
+    }
+}
+
+precommit_master {
+    title = "precommit (master)"
+    cleanup-tasks = ["clean"]
+    tasks = ["precommit"]
+    gradle-args = ["--no-scan", "--no-build-cache"]
+    run-using = cli // value can be "cli" or "tooling-api"
+    daemon = warm // value can be "warm", "cold", or "none"
+    warm-ups = 5
+    iterations = 10
+    git-checkout = {
+        build = "master"
+    }
+}
+
+single_project_branch {
+    title = "single project (@testGitCommit@)"
+    cleanup-tasks = [":server:clean"]
```

Comment:
Ideally we would test an ABI or non ABI change here. But the gradle profiler generated code for that, generates invalid formatted code (spotless and checkstyle complains). If we really want that we should fix that in the gradle profiler and provide a PR at gradle/gradle-profiler to make this work in our cases

Reply:
Could we just run `spotlessApply` first?

Reply:
Unfortunately we cannot invoke spotlessApply as part of the clean run as the abi change is done between the cleanup run and the benchmark run. If we include spotlessApply in the benchmark build it. 
For now I've added spotlessApply to the benchmark build. Since gradle is not yet running tasks in parallel within a project  `spotlessApply` should always be executed first without the need to tweak anything in our existing build logic. This tinkers a bit with the actual scenario as we usually never invoke `:server:spotlessApply :server:precommit` but that should probably be fine for now.